### DISCUSSION TITLE
Newer FactoryBot?

### DISCUSSION THREAD

**Discussion by tmtrademarked** [solidusio/solidus#4254] [New Features or Ideas]
It looks like, today, Solidus sort of implicitly requires FactoryBot 4.8. That version is from 2017, so is getting pretty ancient at this point. I think as Ruby 3.1 comes into mainstream, I think there are changes coming to FactoryBot that might be nice to enable for the Solidus community. (Nothing specific, but happened to be reading through some FactoryBot changelogs recently!)

It's not clear to me how much work is involved in changing the "default recommended" version of FactoryBot in Solidus, though. I know there have been discussions around the corner cases that might break, etc, but it's not obvious to me what they would be expected to be.

Is there any interest in this generally? Has anyone done any thinking about how to bring FactoryBot up to a more modern version? Any known dragons to be careful with here?

**Comment by jarednorman** [comment]
In what way are we implicitly dependent on FactoryBot 4.8? I'm running 6.2 on the app I'm currently most active working on.

**Reply by tmtrademarked** [reply]
https://github.com/solidusio/solidus/blob/master/core/lib/spree/testing_support/factory_bot.rb#L47 - if you use a version other than 4.8, you get a warning in your console about possible strange side effects. To me, that seems to indicate that running 4.8 is the "supported" configuration of Solidus. Is that not a reasonable inference?

**Reply by jarednorman** [reply]
I just opened #4255 to address this.
